frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -304,91 +302,58 @@
-                                                          update-unames!
-                                                          ids-map
-                                                          child
-                                                          delta
-                                                          new-id
-                                                          new-id))
-                        changes
-                        (map (d/getf objects) (:shapes obj)))]
-    changes))
-
 (defn- prepare-duplicate-shape-change
-  [changes objects page unames update-unames! ids-map obj delta frame-id parent-id]
-  (if (some? obj)
-    (let [new-id      (ids-map (:id obj))
-          parent-id   (or parent-id frame-id)
-          name        (:name obj)
-
-          new-obj     (-> obj
-                          (assoc :id new-id
-                                 :name name
-                                 :parent-id parent-id
-                                 :frame-id frame-id)
-                          (dissoc :shapes
-                                  :main-instance?)
-                          (gsh/move delta)
-                          (d/update-when :interactions #(ctsi/remap-interactions % ids-map objects)))
-
-          changes (-> (pcb/add-object changes new-obj {:ignore-touched true})
-                      (pcb/amend-last-change #(assoc % :old-id (:id obj))))]
-
-      (reduce (fn [changes child]
-                (prepare-duplicate-shape-change changes
-                                                objects
-                                                page
-                                                unames
-                                                update-unames!
-                                                ids-map
-                                                child
-                                                delta
-                                                frame-id
-                                                new-id))
-              changes
-              (map (d/getf objects) (:shapes obj))))
-    changes))
+  ([changes objects page unames update-unames! ids-map obj delta]
+   (prepare-duplicate-shape-change changes objects page unames update-unames! ids-map obj delta (:frame-id obj) (:parent-id obj)))
+
+  ([changes objects page unames update-unames! ids-map obj delta frame-id parent-id]
+   (if (some? obj)
+     (let [frame?      (cph/frame-shape? obj)
+           new-id      (ids-map (:id obj))
+           parent-id   (or parent-id frame-id)
+           name        (:name obj)
+
+           new-obj     (-> obj
+                           (assoc :id new-id
+                                  :name name
+                                  :parent-id parent-id
+                                  :frame-id frame-id)
+                           (dissoc :shapes
+                                   :main-instance?)
```

Comment:
think you should also remove `:use-for-thumbnail?`, don't you?

Reply:
ok